js/src/components/RoundedRectangleGraphNode.vue

```diff
@@ -0,0 +1,264 @@
+<template>
+  <g>
+      <rect :width="size.rx" :height="size.ry"
+            :x="-size.rx/2" :y="-size.ry/2"
+            :fill="fill" :stroke="stroke" :stroke-width="strokeWidth" :rx="hover ? 30 : 25"></rect>
+
+      <text id="text" :font-size="textSize" ref="text" x="0" :y="subtext != null ? -8 : 0" :fill="textColour" text-anchor="middle" alignment-baseline="middle">
+      {{displayText.text}}
+    </text>
+    <text id="subtext" :font-size="textSize"  v-if="subtext != null" ref="subtext" x="0" y="8" :fill="textColour" text-anchor="middle" alignment-baseline="middle">
+      {{displayText.subtext}}
+    </text>
+    <title>{{text}}</title>
+  </g>
+</template>
+
+<script lang="ts">
+import Vue, { ComponentOptions } from "vue";
+import Component from "vue-class-component";
+import { Prop, Watch } from "vue-property-decorator";
+
+/**
+ * An elliptical node that supports two lines of text, with automatic truncation and resizing.
+ * 
+ * Events Emitted:
+ * - 'updateBounds':
+ *       The bounds (size of node) have be    en updated.
+ *       Passes a size object {rx: number, ry: number} as an argument.
+ *       To correctly implement automatic node sizing, you should listen for this event and update
+ *       the edges to use this information. E.g. for DirectedEdge, update (source|dest)R(x|y) prop.
+ */
+@Component
+export default class RoundedRectangleGraphNode extends Vue {
+  /** The primary text of the node to display. */
+  @Prop() text: string;
+  /** The secondary text of the node to display. Optional. Only used if a non-null value is provided */
+  @Prop({ required: false })
+  subtext: string;
+  /** A HTML colour string used for the node's fill colour. */
+  @Prop({ default: "white" })
+  fill: string;
+  /** A HTML colour string used to outline the node. */
+  @Prop({ default: "black" })
+  stroke: string;
+  /** The width of the stroke to draw around the node. */
+  @Prop({ default: 1 })
+  strokeWidth: number;
+  /** The colour of the (sub)text inside the node. */
+  @Prop({ default: "black" })
+  textColour: string;
+  // The size of the text inside the node
+  @Prop({default: 10}) textSize: number;
+  /** Flag to see if the node is hovered over */
+  @Prop({default: false}) hover: boolean
+
+  /** The final width, in pixels, of the text element containing the (truncated) text. */
+  computedTextWidth = 0;
+  /** The final width, in pixels, of the subtext element containing the (truncated) subtext. */
+  computedSubtextWidth = 0;
+  /** The total height, in pixels, that the text and subtext (if available) take up. */
+  computedTotalHeight = 0;
+  /** The radius along the x-axis of the ellipse. Will be updated whenever the text changes. */
+  rx = 0;
```

Comment:
Since this class is no longer used to create ellipses, perhaps we can now get rid of rx and ry and use width and height parameters like in the RectangleGraphNode. 

This class also seems like it could share a base class with RectangleGraphNode since the two classes do almost the same thing.

Reply:
done.